zebra-chain/src/primitives/zcash_note_encryption.rs

```diff
@@ -0,0 +1,68 @@
+//! Contains code that interfaces with the zcash_note_encryption crate from
+//! librustzcash.
+//!
+use crate::{
+    block::Height,
+    parameters::{Network, NetworkUpgrade},
+    primitives::zcash_primitives::convert_tx_to_librustzcash,
+    transaction::Transaction,
+};
+
+/// Returns true if all Sapling or Orchard outputs, if any, decrypt successfully with
+/// an all-zeroes outgoing viewing key.
+///
+/// # Panics
+///
+/// If passed a network/height without matching consensus branch ID (pre-Overwinter),
+/// since `librustzcash` won't be able to parse it.
+pub fn decrypts_successfully(transaction: &Transaction, network: Network, height: Height) -> bool {
```

Comment:
Ideally this would all live in `zebra-consensus` until we implement our full note encryption/decryption natively in `zebra-chain`, but since it requires `zcash_primitives` too, pulling in `orchard` into `zebra-chain` is ok for now I'd say (I'm also pulling it in to `zebra-consensus` to do the Halo2 proof verification of the Orchard Actions).

So I guess I'm saying this is fine for now, I hope we refactor it away pretty soon. 🤞

Reply:
Yeah I wasn't sure where to place this, I tried to follow the same logic as `zcash_primitives.rs` which has a similar structure.